test/explicit-resource-management/main.test.ts

```diff
@@ -0,0 +1,123 @@
+
+import { expect } from 'chai';
+import { describe, it } from 'mocha';
+import { AbstractCursor, ChangeStream, ClientSession, GridFSBucket, MongoClient } from 'mongodb/lib/beta';
+import * as sinon from 'sinon';
+import { Readable } from 'stream';
+import { pipeline } from 'stream/promises';
+import { setTimeout } from 'timers/promises';
+
+// @ts-expect-error Assigning readonly property.
+Symbol.asyncDispose ??= Symbol('dispose');
+
+async function setUpCollection(client: MongoClient) {
+	const collection = client.db('foo').collection<{ name: string }>('bar');
+	const documents: Array<{ name: string }> = Array.from({ length: 5 }).map(i => ({
+		name: String(i)
+	}));
+	await collection.insertMany(documents)
+	return collection;
+}
+
+describe('explicit resource management smoke tests', function () {
```

Comment:
I don't think that's necessary, the contents of that function are straightforward.  This is intended as a smoke test, not exhaustive testing.

Reply:
Since we have two copies of the dispose functions, and possibly may add more (thinking txn), is there a way we can assess that the functions are identical?